code/modules/mob/inventory.dm

```diff
@@ -128,15 +128,25 @@
 	Arguments
 	* obj/item/W is the item you are trying to equip
 	* del_on_fail if true will delete the item instead of dropping it to the floor
+	* merge_stacks if true will merge stacks that src is holding in its hands with new ones
 
 	Returns TRUE if it was able to put the thing into one of our hands.
 */
-/mob/proc/put_in_hands(obj/item/W, del_on_fail = FALSE)
+/mob/proc/put_in_hands(obj/item/W, del_on_fail = FALSE, merge_stacks = TRUE)
 	W.do_pickup_animation(src)
 	if(status_flags & INCORPOREAL) // INCORPOREAL things don't have hands
 		return FALSE
 	if(!W)
 		return FALSE
+	if(merge_stacks && istype(W, /obj/item/stack))
```

Comment:
There are few reasons why I put it here:
1. Mob already has a proc called put_in_hands which selects where to put the item.
2. You have to check if stacks can merge before failing and putting item on the ground.
3. In the proc that handles interaction between mob and item you will always have to typecheck one or the other.
4. I don't want to make new procs just for this simple change. More procs = less readable code.

More about "snowflake" and "smelly" code. Those definitions are subjective, so I don't understand what you mean. You looked at the code and it felt "not right". I don't understand what is wrong with it because I do not have your experience.

I don't like where this conversation is going, I think I spent more energy trying to explain the change than actually making the change. And English is not my first language.
I wanted a simple fix to make people life better, and you instead of helping me do the fix, just point and say "that's wrong", "this is not how you do that". I know. You would have done it different;y. If you don't like it, close the PR.

Reply:
You've had 3 maints tell you the same thing, we are helping you fix it, we're telling you where to move the code.

Its snowflake because it is stack item specific code, being handled entirely by a mob level proc which is supposed to be very generic. You generally want things handled as internally as possible, its more maintainable, its cleaner, and easier to read.

What happens if we have another pick up interaction with another item type? Do we add more snowflake for that type? What about a 3rd? a 10th? What happens if stack item code changes in the future, or something else shares behavior but isn't the same type?

Reply:
"I don't like where this conversation is going, I think I spent more energy trying to explain the change than actually making the change."

I get your feeling, your code is working. But you said it yourself, you are not familiar with the code base. We are pointing out that here, we want out mob code to be extra generic. Which means the logic for stacking item should be on /obj/item/stack. 

And yes, "snowflake" is very subjective, but this looks to be a consensus among maints.

"I wanted a simple fix to make people life better, and you instead of helping me do the fix, just point and say "that's wrong", "this is not how you do that". I know. You would have done it different;y. If you don't like it, close the PR." 
I think we are trying to help here, by pointing out where we think the code should be. 
Your feature seems interesting, and I would really like not to close this PR; but you also gotta play the open source game, which is to make the maintainer like your code.

Please try to put your logic in stack, ask for help in discord if needs be, and this will be merged with no problems